lib/ansible/modules/cloud/lxd/lxd_container.py

```diff
@@ -458,7 +458,7 @@ def _needs_to_change_container_config(self, key):
         if key == 'config':
             old_configs = dict((k, v) for k, v in self.old_container_json['metadata'][key].items() if not k.startswith('volatile.'))
             for k, v in self.config['config'].items():
-                if old_configs[k] != v:
+                if k in old_configs and old_configs[k] != v:
```

Comment:
It's not clear what should be the right course of action here: if a key is in the new config and not in old, that's probably also a change, but you're skipping it.
Maybe have something like this?
```python
if k not in old_configs or old_configs[k] != v:
```

Reply:
You're right, except that I don't think we want to propagate the volatiles at that stage. We need to see what the initial dev expected here.